packages/react-client/src/ReactFlightClient.js

```diff
@@ -641,7 +641,15 @@ function createElement(
     }
     // TODO: We should be freezing the element but currently, we might write into
     // _debugInfo later. We could move it into _store which remains mutable.
-    Object.freeze(element.props);
+    if (
+      initializingChunkBlockedModel !== null &&
+      initializingChunkBlockedModel.deps > 0
+    ) {
+      const freeze = Object.freeze.bind(Object, element.props);
+      initializingChunk.then(freeze, freeze);
```

Comment:
Q: is there a possibility that this callback would run before a `createModelResolver` callback, in which case we'd still mutate a frozen object? or is the inside-out traversal order of the JSON reviver protecting us from that

Reply:
This waits until the entire chunk that was being parsed is resolved which means that everything inside it has to be initialized already. Otherwise other user spaces listeners could observer the intermediate state.

However, this does mean that an early listener to this chunk can observe it before we freeze it. Which is actually quite likely since initialization happens lazily in the first place. But this is an edge case that you need to wait for the blocked model in the first place and it's dev only and you might end up hitting the freeze later if you do something similar anyway.